streams/src/commonMain/kotlin/com/mirego/trikot/streams/reactive/BehaviorSubjectImpl.kt

```diff
@@ -10,7 +10,9 @@ open class BehaviorSubjectImpl<T>(
 
     override fun onNewSubscription(subscription: PublisherSubscription<T>) {
         super.onNewSubscription(subscription)
-        this.value?.let { subscription.dispatchValue(it) }
-        this.error?.let { subscription.dispatchError(it) }
+        serialQueue.dispatch {
```

Comment:
Please remove this change. This will dispatch the value on another run loop when its already set. Probably in an async way. This code was in place so if the publisher is completed you receive the value sync and not store your subscription token.